src/ssa.c

```diff
@@ -625,14 +626,25 @@ void append_unwound_phi_insn(basic_block_t *bb, var_t *dest, var_t *rs)
     n->rd = dest;
     n->rs1 = rs;
 
+    insn_t *head = bb->insn_list.head;
     insn_t *tail = bb->insn_list.tail;
     if (!tail) {
         bb->insn_list.head = n;
         bb->insn_list.tail = n;
     } else {
         /* insert it before branch instruction */
         if (tail->opcode == OP_branch) {
-            if (tail->prev) {
+            if (head->opcode == OP_phi) {
+                /* FIXME: This temporary implementation handles the case that if
+                 * tail->prev is OP_phi. The error in insert_phi_insn() is due
+                 * to not linking the `prev` pointer of the next instruction to
+                 * the newly added phi instruction. Once this error is fixed, we
+                 * can remove the first if statement.
+                 */
+                while (head->next->opcode != OP_branch)
```

Comment:
Can you avoid traversing the whole elements since we just want to figure out the branch operation at first glance?

Reply:
Should I still modify this part of the code, or should I keep the current commit as is and wait for @vacantron 's fix?

Reply:
You can cherry-pick the commit at https://github.com/vacantron/shecc/commit/5fd2fbe211cc9bf49a84fb768169b38ea841859c and update your change.